compiler.ts

```diff
@@ -399,43 +401,39 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
         var brObjStmts = codeGenExpr(expr.obj, env);
         var brKeyStmts = codeGenExpr(expr.key, env);
         var brStmts = [];
-        //First check whether index is out of bounds, if so, throw error by going to print_str. Then index the string.
         brStmts.push(
           ...[
             `${brObjStmts.join("\n")}`, //Load the string object to be indexed
+            `(local.set $$string_address)`,
```

Comment:
What happens if `brKeyStmts` was compiled from an expression that also compiles to set `$$string_address`?

For example `a[f(b[5])]`?